packages/core/usePublicClassNames/usePublicClassNames.js

```diff
@@ -0,0 +1,37 @@
+import {mergeClassNameMaps} from '../util';
+
+/**
+ * A hook for supporting `publicClassNames` to functional components.
+ * It returns merged CSS of given two CSS objects according to `publicClassNames` option.
+ *
+ * @param   {Object.<string, string>}    [componentCss]     The CSS of the component
+ * @param   {Object.<string, string>}    [customCss]        The supplied collection of CSS class names to the
+ *                                                          corresponding internal elements and states of the component
+ * @param   {Boolean|String|String[]}    [publicClassNames] The keys of public class names of the component
+ *                                                          If this value is `true`, all of the keys from the component
+ *                                                          CSS will become public class names.
+ * @returns {Object}                                        A merged CSS
+ * @private
+ */
+function usePublicClassNames ({componentCss, customCss, publicClassNames}) {
+	let allowedClassNames = publicClassNames;
+	let mergedCss = componentCss;
+
```

Comment:
How about skip the unnecessary logic here?

```suggestion

	if (!customCss) {
		return mergedCss;
	}
```
and remove L27 condition "customCss"